noir-projects/aztec-nr/aztec/src/context/call_interfaces.nr

```diff
@@ -467,13 +467,13 @@ impl<let N: u32> CallInterface<N> for PublicStaticVoidCallInterface<N> {
 }
 
 pub struct PublicStaticVoidCallInterface<let N: u32> {
-    target_contract: AztecAddress,
-    selector: FunctionSelector,
-    name: str<N>,
-    args: [Field],
-    return_type: (),
-    is_static: bool,
-    gas_opts: GasOpts,
+    pub target_contract: AztecAddress,
+    pub selector: FunctionSelector,
+    pub name: str<N>,
+    pub args: [Field],
+    pub return_type: (),
+    pub is_static: bool,
+    pub gas_opts: GasOpts,
```

Comment:
This was causing macro warnings in Router contract because there we instantiated the struct as `PublicStaticVoidCallInterface{ target_contract, selector, ... }`.

Reply:
You have a point. But all the other 5 structs there also have it exposed so I decided to the change in a PR up the stack. I will also doing a related refactor that since I found it all incredibly hard to read.